processing/src/main/java/io/druid/segment/data/CompressedLongsIndexedSupplier.java

```diff
@@ -73,302 +58,57 @@ public int size()
   @Override
   public IndexedLongs get()
   {
-    final int div = Integer.numberOfTrailingZeros(sizePer);
-    final int rem = sizePer - 1;
-    final boolean powerOf2 = sizePer == (1 << div);
-    if(powerOf2) {
-      return new CompressedIndexedLongs() {
-        @Override
-        public long get(int index)
-        {
-          // optimize division and remainder for powers of 2
-          final int bufferNum = index >> div;
-
-          if (bufferNum != currIndex) {
-            loadBuffer(bufferNum);
-          }
-
-          final int bufferIndex = index & rem;
-          return buffer.get(buffer.position() + bufferIndex);
-        }
-      };
-    } else {
-      return new CompressedIndexedLongs();
-    }
+    return supplier.get();
   }
 
   public long getSerializedSize()
   {
-    return baseLongBuffers.getSerializedSize() + 1 + 4 + 4 + 1;
+    return buffer.remaining();
   }
 
   public void writeToChannel(WritableByteChannel channel) throws IOException
   {
-    channel.write(ByteBuffer.wrap(new byte[]{version}));
-    channel.write(ByteBuffer.wrap(Ints.toByteArray(totalSize)));
-    channel.write(ByteBuffer.wrap(Ints.toByteArray(sizePer)));
-    channel.write(ByteBuffer.wrap(new byte[]{compression.getId()}));
-    baseLongBuffers.writeToChannel(channel);
+    channel.write(buffer.asReadOnlyBuffer());
   }
 
   public CompressedLongsIndexedSupplier convertByteOrder(ByteOrder order)
```

Comment:
It looks like this is only called by MetricHolder.convertByteOrder, which nothing calls. let's just remove both.